File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        int N = 9;
       HashSet<Character>[] rows = new HashSet[9];
       HashSet<Character>[] cols = new HashSet[9];
       HashSet<Character>[] boxes = new HashSet[9];


       for (int i = 0; i < N; i++) {
           rows[i] = new HashSet<>();
           cols[i] = new HashSet<>();
           boxes[i] = new HashSet<>();
           
       }
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                char c = board[i][j];
                if (c !=  '.') {
                    if (!rows[i].add(c) || !cols[j].add(c) || !boxes[i/3*3 + j/3].add(c)) {
                        return false;
                    }
                }
            }
        }
        return true;
    }
}

File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        unordered_set<string> my_set;

        for (int i = 0; i < board.size(); i++) {
            for (int j = 0; j < board[0].size(); j++) {
                if (board[i][j] != '.') {
                    string s = "(" + string(1, board[i][j]) +")";
                    if (my_set.find(to_string(i) + s) == my_set.end() && my_set.find(to_string(i/3) + s + to_string(j/3)) == my_set.end() && my_set.find(s + to_string(j)) == my_set.end()) {
                        my_set.insert(to_string(i) + s);
                        my_set.insert(to_string(i/3) + s + to_string(j/3));
                        my_set.insert(s + to_string(j));
                    } else {
                        return false;
                    }
                }
            }
        }

        return true;
    }
};